mypy/build.py

```diff
@@ -897,23 +962,33 @@ def find() -> Optional[str]:
             # Prefer package over module, i.e. baz/__init__.py* over baz.py*.
             for extension in PYTHON_EXTENSIONS:
                 path = base_path + sepinit + extension
+                path_stubs = base_path + '-stubs' + sepinit + extension
                 if is_file(path) and verify_module(id, path):
                     return path
+                elif is_file(path_stubs) and verify_module(id, path_stubs):
+                    return path_stubs
             # No package, look for module.
             for extension in PYTHON_EXTENSIONS:
                 path = base_path + extension
                 if is_file(path) and verify_module(id, path):
                     return path
         return None
 
-    key = (id, lib_path)
-    if key not in find_module_cache:
-        find_module_cache[key] = find()
-    return find_module_cache[key]
+    if id not in find_module_cache:
+        find_module_cache[id] = find()
+
+    # If we searched for items with a base directory of site-packages/ we need to
+    # remove it to avoid searching it for non-typed ids.
+    for dir in site_packages_dirs:
+        if dir + os.sep in find_module_dir_cache[dir_chain]:
+            find_module_dir_cache[dir_chain].remove(dir + os.sep)
```

Comment:
Can this be moved inside `find`, since that's the only place that sets `find_module_dir_cache`?

Reply:
Not easily, `find` has several return paths, and I need to clean this up for all of them.

Reply:
It can at least be moved inside the above `if` then, which only runs if `find()` ran